[CBRD-23840] Separation/Integration of cci repositories from CUBRID repository.

hwany7seo opened the pull request:
http://jira.cubrid.org/browse/CBRD-23949

Purpose
- Added git hash tag for packaging in engine build.
Implementation
N/A
Remarks
N/A

hwany7seo committed:
[CBRD-23840] Separation/Integration of cci repositories from CUBRID repository.
http://jira.cubrid.org/browse/CBRD-23949

- Added git hash tag for packaging in engine build.

kisoo-han commented on cci/CMakeLists.txt:
```diff
@@ -109,13 +112,21 @@ else(VERSION_MATCHES_LENGTH GREATER 3)
+    if(git_result)
+      message(FATAL_ERROR "Could not get hash information from Git")
+    endif(git_result)
+    set(CCI_HASH_TAG -${commit_hash})
```
What about this (removing '-' from CCI_HASH_TAG, and put '-' when we make **BUILD NUMBER**)
Anyway, results are same, and it is trivial.

`set(CCI_HASH_TAG ${commit_hash})`

kisoo-han commented on cci/CMakeLists.txt:
```diff
@@ -109,13 +112,21 @@ else(VERSION_MATCHES_LENGTH GREATER 3)
   endif(EXISTS "${CCI_ROOT_PATH}/.git" AND GIT_FOUND)
 endif(VERSION_MATCHES_LENGTH GREATER 3)
-set(CCI_BUILD_NUMBER ${CCI_MAJOR_VERSION}.${CCI_MINOR_VERSION}.${CCI_PATCH_VERSION}.${CCI_EXTRA_VERSION})
+set(CCI_BUILD_NUMBER ${CCI_MAJOR_VERSION}.${CCI_MINOR_VERSION}.${CCI_PATCH_VERSION}.${CCI_EXTRA_VERSION}${CCI_HASH_TAG})
```
And,
set(CCI_BUILD_NUMBER ${CCI_MAJOR_VERSION}.${CCI_MINOR_VERSION}.${CCI_PATCH_VERSION}.${CCI_EXTRA_VERSION}-${CCI_HASH_TAG})

kisoo-han commented:
Even it is not the scope of this change, I have a question on your previous modification on CMakeLists.txt
Is it same meaning? Then what about change it to the second style. Because, single build of cci from cci-repo,
the value of the EP_TARGETS is "libopenssl", single value. However, if it is called from engine build the value of the
EP_TARGETS will be "libopenssl" "libexpat" "..." and 1st style if could be failed. (I tried it)

```
if(NOT ${EP_TARGETS} STREQUAL "")
    add_dependencies(cascci_static ${EP_TARGETS})
endif(NOT ${EP_TARGETS} STREQUAL "")
```

```
if(DEFINED EP_TARGETS)
    add_dependencies(cascci_static ${EP_TARGETS})
endif(DEFINED EP_TARGETS)
```

kisoo-han reviewed (approved):
Regarding location of cci_applier, engine or cci, we could postpone the discussion to the future.
As I told you, please PR engine part for CCI for review.

hwany7seo committed:
changed After Review
- not include special characters in 'HASH_TAG' variable.
- changed 'EP_TARGETS' variable check method.

hwany7seo commented:
> Regarding location of cci_applier, engine or cci, we could postpone the discussion to the future.
> As I told you, please PR engine part for CCI for review.

changed as per your review comment. and engine part will be PR soon.
